### DISCUSSION TITLE
Make an all in one solution for ReplayBook

### DISCUSSION THREAD

**Discussion by icecoffeeabell1835** [fraxiinus/ReplayBook#102] [Feature Request]
Hello all,

I have used ReplayBook for the first time.

1. First I loaded the ReplayBook 1.3.2 file.
2. Then immediately noticed during setup that I can download more things, like.
- Download champion thumbnails
- Download item thumbnails
- Download rune thumbnails

3. After downloading, I copied my saved Replays files (`.rofl`) to the default path. `C:\Users\x\Documents\League of Legends\Replays`
4. Then I realized that I can't view my replays because I don't have the older client version. The first experience was very user unfriendly and terrible.

My suggestion is that it should be created an all in one solution for ReplayBook. That means when I download and install the ReplayBook file, everything I wrote in step 2 is already integrated by default.

In addition, it should be in ReplayBook the ability to download the older client version directly, the location can also select the user directly in the UI or drag & drop it add.

**Comment by fraxiinus** [comment]
Hi icecoffeeabell1835,
You bring up a good point that the first-time user setup is not streamlined. Right now, the setup provides an option to skip the downloading step entirely, that way users wouldn't have to sit waiting for downloads. ReplayBook would then download images on-demand. I also make it clear that ReplayBook can only play a replay if you have the matching client. You should still be able to view players, stats, and runes.
I've been thinking of how to improve this experience, and I think the following actions would help:

* Download images in a large zip file, instead of one file at a time. Increases download speed
* As you said: Provide an "complete" download of ReplayBook that includes everything in one zip file.

I'll see what I can do to implement the above in the future.

As for your second suggestion, I'm going to say outright that I would never be able to provide such a feature. Not only would programming something that downloads and manages 8+ GB files be troublesome, but the data hosting costs would be immense. For a free open-source tool, my funds are limited and I can't justify spending that kind of money on that.

However, many users have suggested that ReplayBook should have an auto-backup feature. So when a new patch releases, you don't have to worry about backing up your game manually. I have been hesitant on starting this feature, but it will probably be part of ReplayBook one day.

**Reply by icecoffeeabell1835** [reply]
> As for your second suggestion, I'm going to say outright that I would never be able to provide such a feature. 

Maybe it is possible with this tool and you can create something similar or integrate it directly to ReplayBook.
https://github.com/LoL-Fantome/LeagueDownloader

I have not tested it but according to the description any client version of Riot Server can be downloaded.

**Reply by fraxiinus** [reply]
I wanted to do that 😢 but I heard that it doesn't work anymore since Riot moved away from RADS

**Reply by icecoffeeabell1835** [reply]
> 
> 
> I wanted to do that cry but I heard that it doesn't work anymore since Riot moved away from RADS

This is terrible. 😢

But is it generally still possible to download any client version of Riot Server with another tool that supports the new servers or has Riot removed it completely?

**Reply by fraxiinus** [reply]
No idea... I last did research into this, maybe 2 years ago. I'll have to look into it again.
